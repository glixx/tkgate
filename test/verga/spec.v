module top;
  integer i;
  reg a,b,c;
  wire x,y,z;

  initial
    $monitor("%4d: a=%b b=%b c=%b - x=%b y=%b z=%b",$time,a,b,c,x,y,z);

  initial
    for (i = 0;i < 8;i = i + 1)
      begin
	{a,b,c} = i;
	#50;
//	$finish();
      end

  foo f(a,b,c,x,y,z);

endmodule

module foo(a,b,c,x,y,z);
input a,b,c;
output x,y,z;
wire q,r;

  specify
    (a,b *> x) = 12;
    (c *> x) = 8;
    (a *> y) = 11;
    (c *> y) = 16;
    (b *> z) = 23;
    (c *> z) = 18;
  endspecify

//  and (r, q, c);
//  xor (x, a, b, r);

  assign r = q & c;
  assign x = a ^ b ^ r;
  assign y = a & c;
  assign z = c & b;
  assign q = a & b;

endmodule
