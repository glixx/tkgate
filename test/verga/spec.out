   0: a=0 b=0 c=0 - x=z y=z z=z
   8: a=0 b=0 c=0 - x=x y=z z=z
  11: a=0 b=0 c=0 - x=x y=0 z=z
  12: a=0 b=0 c=0 - x=0 y=0 z=z
  18: a=0 b=0 c=0 - x=0 y=0 z=0
  50: a=0 b=0 c=1 - x=0 y=0 z=0
 100: a=0 b=1 c=0 - x=0 y=0 z=0
 112: a=0 b=1 c=0 - x=1 y=0 z=0
 150: a=0 b=1 c=1 - x=1 y=0 z=0
 168: a=0 b=1 c=1 - x=1 y=0 z=1
 200: a=1 b=0 c=0 - x=1 y=0 z=1
 211: a=1 b=0 c=0 - x=1 y=1 z=1
 216: a=1 b=0 c=0 - x=1 y=0 z=1
 218: a=1 b=0 c=0 - x=1 y=0 z=0
 250: a=1 b=0 c=1 - x=1 y=0 z=0
 266: a=1 b=0 c=1 - x=1 y=1 z=0
 300: a=1 b=1 c=0 - x=1 y=1 z=0
 312: a=1 b=1 c=0 - x=0 y=1 z=0
 316: a=1 b=1 c=0 - x=0 y=0 z=0
 350: a=1 b=1 c=1 - x=0 y=0 z=0
 358: a=1 b=1 c=1 - x=1 y=0 z=0
 366: a=1 b=1 c=1 - x=1 y=1 z=0
 368: a=1 b=1 c=1 - x=1 y=1 z=1
